lib/scout_apm/agent_context.rb

```diff
@@ -9,6 +12,7 @@ class AgentContext
     def initialize()
       @logger = LoggerFactory.build_minimal_logger
       @process_start_time = Time.now
+      @extensions = ScoutApm::Extensions::Config.new
```

Comment:
If you pass `self` into `initialize`, you can then reference that local variable later on, rather than reaching back out to the global `AgentContext`

Reply:
Currently the interface for adding and running extensions is via class methods, not instance methods. With the current design, it feels like it'd be a little confusing to set a class var in `initialize`.